project/merge.go

```diff
@@ -50,8 +50,21 @@ func Merge(p *Project, bytes []byte) (map[string]*ServiceConfig, error) {
 		datas[name] = data
 	}
 
-	err = utils.Convert(datas, &configs)
-	return configs, err
+	if err := utils.Convert(datas, &configs); err != nil {
+		return nil, err
+	}
+
+	adjustValues(configs)
+	return configs, nil
+}
+
+func adjustValues(configs map[string]*ServiceConfig) {
+	// yaml parser turns "no" into "false" but that is not valid for a restart policy
+	for _, v := range configs {
+		if v.Restart == "false" {
+			v.Restart = "no"
+		}
+	}
```

Comment:
No, the yaml parser isn't wrong, the user is wrong.

If you go that route, you end up translating a bunch of things that aren't `no` into `"no"`.

Using a boolean in a string field should result in an error. The user should write `restart: 'no'`